main.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times { |i| yield(to_a[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    size.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    ary = []
+    my_each do |elem|
+      ary << elem if yield elem
+    end
+    ary
+  end
+
+  def my_all?
+    return nil unless block_given?
+
+    my_each do |elem|
+      return false unless yield elem
+    end
+    true
+  end
```

Comment:
- enumerables #my_all? when no block or argument is given returns true when none of the collection members are false or nil
- enumerables #my_all? when no block or argument is given returns false when one of the collection members are false or nil
- enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Integer
- enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric
- enumerables #my_all? when a class is passed as an argument returns false if any of the collection is not a member of such class
- enumerables #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- enumerables #my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
- enumerables #my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
- enumerables #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern